src/isDeepEqual.ts

```diff
@@ -64,6 +64,29 @@ export function isDeepEqual(): unknown {
   return purry(isDeepEqualImplementation, arguments);
 }
 
+function isDeepEqualMaps(
+  data: Readonly<Map<unknown, unknown>>,
+  other: Readonly<Map<unknown, unknown>>,
```

Comment:
This doesn't make the map readonly, maps have their own readonly type like arrays do:

```suggestion
  data: ReadonlyMap<unknown, unknown>,
  other: ReadonlyMap<unknown, unknown>,
```